### PR TITLE
Add flag to tradeoff error vs. performance

### PR DESCRIPTION
**Pull request by cpuhrsch**

### CODE REVIEW

File: protoquant/src/triton/quant.py [hunk 1]
```diff
@@ -104,28 +104,29 @@ def quant_kernel(inputs):
...
+#    nudged_zero_point = torch.where(initial_zero_point > qmax, qmax, nudged_zero_point)
 
     inv_scale = 1.0 / scale
     precomputed_sum = torch.round(row_sum_val * inv_scale).to(torch.int32)
 
-    transformed_val = (inputs * inv_scale.unsqueeze(1)).to(torch.float32)
-    transformed_val = torch.round(transformed_val) + nudged_zero_point.unsqueeze(1)
+    transformed_val = (inputs * inv_scale.unsqueeze(1)).to(torch.int32) #float32)
+    # transformed_val = torch.round(transformed_val) + initial_zero_point.unsqueeze(1)
+    transformed_val = transformed_val + initial_zero_point.unsqueeze(1)
```
Comment (cpuhrsch) [inline protoquant/src/triton/quant.py]: I'm wondering if doing this in float32 instead of int32 might help performance. I'm still playing a bit with the type conversions to avoid calls to `round` etc.

File: protoquant/src/triton/quant.py [hunk 2]
```diff
@@ -104,28 +104,29 @@ def quant_kernel(inputs):
...
 
     output = torch.clamp(transformed_val, min=qmin, max=qmax).to(torch.int8)
```
Comment (nmacchioni) [inline protoquant/src/triton/quant.py]: I think we can find a way to avoid `torch.clamp`, since its only purpose is to avoid overflow or underflow. Maybe we can calculate the scales based on a range of only 255, not 256. Lose a tiny bit of quantization range for one less call.

File: protoquant/src/triton/quant.py (hunk 1 above)
Comment (nmacchioni) [reply protoquant/src/triton/quant.py]: I think we should be able to move the shifts to float32, given that we aren't doing the zero nudging anymore.

**Review by nmacchioni — APPROVED** [review]
LGTM

File: protoquant/src/triton/quant.py [hunk 3]
```diff
@@ -104,28 +104,29 @@ def quant_kernel(inputs):
         zero_point_from_min_error < zero_point_from_max_error,
```
Comment (cpuhrsch) [inline protoquant/src/triton/quant.py]: @nmacchioni - I'm also wondering whether this is a decision we could make statically even if it results in a loss of accuracy? Seems like a lot of preceding code (and reductions) depend on this.

Comment (nmacchioni) [reply protoquant/src/triton/quant.py]: We could either do zero point from min or max, and just choose which one to use, or we could take zero point as the midpoint of the min and max. Both ways should be fine.
